### PR TITLE
Allow for attempting to automatically approve of normal cases of charges

### PR DESCRIPTION
**Pull request by vonElfvin**

### CODE REVIEW

File: src/main/kotlin/com/hedvig/accountservice/schedulers/web/ScheduleController.kt [hunk 1]
```diff
@@ -128,6 +133,31 @@ class ScheduleController @Autowired constructor(
...
+    @PostMapping("/approvals/auto")
+    fun attemptAutoApproveChargeAmounts(
+        @RequestParam approvedBy: String
+    ): ResponseEntity<Void> {
+        val memberIdsWaitingForApproval = stateManager.findAllMemberIds(ChargeStatus.SUBSCRIPTION_SCHEDULED_AND_WAITING_FOR_APPROVAL)
+        val currentPeriod = YearMonth.now()
+        memberIdsWaitingForApproval.forEach { memberId ->
+            val monthlyBilling = productPricingService.getMonthlyBilling(memberId, currentPeriod)
+            val accountBalance = accountManager.getCurrentBalance(memberId)
+            if (accountBalance.isEqualTo(monthlyBilling.subscription)) {
```
Comment (meletios) [inline src/main/kotlin/com/hedvig/accountservice/schedulers/web/ScheduleController.kt]: Do we need to take into account the discounts here?

File: src/main/kotlin/com/hedvig/accountservice/schedulers/web/ScheduleController.kt [hunk 2]
```diff
@@ -128,6 +133,31 @@ class ScheduleController @Autowired constructor(
         return ResponseEntity.noContent().build()
     }
 
+    @PostMapping("/approvals/auto")
+    fun attemptAutoApproveChargeAmounts(
+        @RequestParam approvedBy: String
+    ): ResponseEntity<Void> {
+        val memberIdsWaitingForApproval = stateManager.findAllMemberIds(ChargeStatus.SUBSCRIPTION_SCHEDULED_AND_WAITING_FOR_APPROVAL)
```
Comment (meletios) [inline src/main/kotlin/com/hedvig/accountservice/schedulers/web/ScheduleController.kt]: is it better to create a specific method without passing the enum? 

Something like this

```suggestion
        val memberIdsWaitingForApproval = stateManager.findAllMemberIdsWaitingForApproval()
```

File: src/main/kotlin/com/hedvig/accountservice/schedulers/web/ScheduleController.kt (hunk 1 above)
Comment (vonElfvin) [reply src/main/kotlin/com/hedvig/accountservice/schedulers/web/ScheduleController.kt]: The idea is to only automatically approve the members who don't have any discounts, any object etc. Just plain old home insurance.

File: pom.xml [hunk 3]
```diff
@@ -26,7 +26,7 @@
         <jdk>1.8</jdk>
         <kotlin.version>1.3.30</kotlin.version>
         <kotlin.code.style>official</kotlin.code.style>
-        <jackson.version>2.9.9</jackson.version>
+        <jackson.version>2.10.0.pr1</jackson.version>
```
[the remark this answers is not in the document]
Comment (vonElfvin) [reply pom.xml]: Because snyk is whining about our jackson version, the only one that satisfies the security is this unfortunally

**Pull request merged**